### PR TITLE
Feature/simplified frontpage

### PR DESCRIPTION
**Pull request by kksteini**
# What and why?

Created a new content type within contentful that merges together elements unique to the frontpage. This makes managing the content easier, keeps it all in one place and lessens the amount of API calls made to the contentful servers.

# How?

Introduced new queries, models and dto's to the API that deal with a new content type called "Frontpage" in contentful.

### CODE REVIEW

File: apps/web/screens/queries/Frontpage.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import gql from 'graphql-tag'
+
+export const GET_FRONTPAGE_QUERY = gql`
+  query GetFrontpage($input: GetFrontpageInput!) {
+    getFrontpage(input: $input) {
```
Comment (stefanhar) [inline apps/web/screens/queries/Frontpage.ts:5]: Maybe we should add `id` field here to better support cache:
https://www.apollographql.com/docs/react/caching/cache-configuration/#default-identifier-generation

File: libs/api/domains/cms/src/lib/models/frontpage.model.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+  @Field(() => Namespace, { nullable: true })
+  namespace?: Namespace
+
+  @Field(() => [LifeEventPage])
+  lifeEvents?: Array<LifeEventPage>
+}
+
+export const mapFrontpage = ({ fields }: IFrontpage): Frontpage => ({
+  title: fields.title ?? '',
```
Comment (stefanhar) [inline libs/api/domains/cms/src/lib/models/frontpage.model.ts:27]: This should include id for cache

File: libs/api/domains/cms/src/lib/cms.contentful.service.ts [hunk 3]
```diff
@@ -568,6 +569,20 @@ export class CmsContentfulService {
...
+      ['content_type']: 'frontpage',
+      include: 10,
+      order: '-sys.createdAt',
+    }
+
+    const result = await this.contentfulRepository
+      .getLocalizedEntries<types.IFrontpageFields>(lang, params)
+      .catch(errorHandler('getFrontpage'))
+
+    return result.items.map(mapFrontpage)[0]
```
Comment (stefanhar) [inline libs/api/domains/cms/src/lib/cms.contentful.service.ts:583]: Maybe we should add a page identifier field like we do on the "Generic Overview Page" that is more robust

**Review by stefanhar — CHANGES_REQUESTED** [review]

File: apps/web/screens/Home.tsx [hunk 4]
```diff
@@ -44,28 +35,19 @@ import { FRONTPAGE_NEWS_TAG_ID } from '@island.is/web/constants'
...
   const { activeLocale, t } = useI18n()
   const { globalNamespace } = useContext(GlobalContext)
   const n = useNamespace(namespace)
   const gn = useNamespace(globalNamespace)
   const { linkResolver } = useLinkResolver()
 
-  if (!lifeEvents || !lifeEvents.length) {
+  if (!page.lifeEvents || !page.lifeEvents.length) {
     return null
   }
```
[the remark this answers is not in the document]
Comment (kksteini) [reply apps/web/screens/Home.tsx:52]: Agree

**Comment by kksteini** [comment]
New query sanity check: 
image

File: libs/api/domains/cms/src/lib/cms.contentful.service.ts (hunk 3 above)
Comment (kksteini) [reply libs/api/domains/cms/src/lib/cms.contentful.service.ts:583]: May I ask why?
What does that add?

**Review by stefanhar — APPROVED** [review]
LGTM

**Review by eirikurn — APPROVED** [review]
Only reviewed mocks. We should configure the right codeowner for each mock domain.

**Pull request merged**